viper/parser/stmt.py

```diff
@@ -363,14 +363,7 @@ def parse_return(self):
         elif isinstance(sub.typ, ListType):
             sub_base_type = re.split(r'\(|\[', str(sub.typ.subtype))[0]
             ret_base_type = re.split(r'\(|\[', str(self.context.return_type.subtype))[0]
-            if sub_base_type != ret_base_type and sub.value != 'multi':
-                raise TypeMismatchException(
-                    "List return type %r does not match specified return type, expecting %r" % (
-                        sub_base_type, ret_base_type
-                    ),
-                    self.stmt
-                )
-            if sub.location == "memory" and sub.value != "multi":
+            if sub_base_type == ret_base_type and sub.location == "memory" and sub.value != "multi":
```

Comment:
I think having an exception here will still be useful?

Reply:
Agree, is there a way that this Exception no longer matters? Sort of unexplained.